incubator/hnc/internal/reconcilers/object.go

```diff
@@ -666,7 +665,10 @@ func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstruct
 	if err != nil {
 		log.Error(err, "Invalid value")
 	}
-	noneSelector := r.getNoneSelector(log, inst)
+	noneSelector, err := selectors.GetNoneSelector(inst)
+	if err != nil {
+		log.Error(err, "Invalid value")
```

Comment:
Add surface error TODO with the issue link (if it's not linked nearby).

Reply:
I think the log is sufficient surfacing here? The validator talks directly to the user, but the reconciler can't.

Reply:
Do we still need to generate an event here?

Reply:
Oh good point! Yes, we should.

Reply:
The only concern about adding it to metrics is that we need to unset it when it's gone.

What about object errors then? We have them as events now but not in the metrics yet, should we also add that? Maybe not for object events, since it's hard to unset them in metrics.

Reply:
Ok let's skip the metric - just an event will be fine.

I think events show up in Stackdriver as metrics and you can take the first derivative to be notified when something new happens.

Reply:
sg